### ISSUE TITLE
go version

### ISSUE THREAD

**Issue by juukee** [gin-contrib/graceful#6]
```
# github.com/gin-contrib/graceful
../../go/pkg/mod/github.com/gin-contrib/graceful@v0.0.0-20230904152018-e56ed94cd808/graceful.go:109:25: undefined: context.WithCancelCause
../../go/pkg/mod/github.com/gin-contrib/graceful@v0.0.0-20230904152018-e56ed94cd808/graceful.go:138:18: undefined: context.Cause
../../go/pkg/mod/github.com/gin-contrib/graceful@v0.0.0-20230904152018-e56ed94cd808/graceful.go:170:32: undefined: context.WithCancelCause
../../go/pkg/mod/github.com/gin-contrib/graceful@v0.0.0-20230904152018-e56ed94cd808/graceful.go:209:16: undefined: context.Cause
note: module requires Go 1.20
```
 my default version is 1.18, so must update my version?

**Comment by cedric-appdirect** [comment]
Hum, I guess I could drop that. It was nice to have. Is there any reason to use an old Go for you?

**Comment by appleboy** [comment]
fixed in #8
